nixos/lib/testing/nodes.nix

```diff
@@ -24,16 +43,18 @@ let
         [
           ./nixos-test-base.nix
           { key = "nodes"; _module.args.nodes = config.nodesCompat; }
-          ({ config, ... }:
+          ({ config, pkgs, ... }:
             {
               virtualisation.qemu.package = testModuleArgs.config.qemu.package;
+              virtualisation.host.pkgs = hostPkgs;
+              nixpkgs.pkgs = guestPkgs;
```

Comment:
We can't "hardcode" a `pkgs` in the default configuration. The default configuration should support all the features of `misc/nixpkgs.nix`, such as `nixpkgs.config`, `nixpkgs.buildPlatform`, etc - ie NixOS needs to call Nixpkgs.

It also happens to break `pkgsReadOnly`:

```nix
nix-repl> testers.runNixOSTest { node.pkgsReadOnly = true; name = "test"; nodes = { machine = { }; }; }
```

This line needs to be moved to a different entrypoint / entrypoints, such as
- `nixosTests`
- `testers.runNixOSTest`

Reply:
This is still a problem.

I think we should improve the [node.pkgsReadOnly](https://nixos.org/manual/nixos/stable/#test-opt-node.pkgsReadOnly) option so that its default is not `options.node.pkgs.isDefined` but something like `options.node.pkgs.highestPrio != optionDefaultPrio`. That way the test framework can provide a default guest package set without affecting `node.pkgsReadOnly`'s default.